homeassistant/components/ring/__init__.py

```diff
@@ -43,16 +43,15 @@
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Ring component."""
-    if DOMAIN not in config:
-        return True
+    if DOMAIN in config:
 
-    def legacy_cleanup():
-        """Clean up old tokens."""
-        old_cache = Path(hass.config.path(".ring_cache.pickle"))
-        if old_cache.is_file():
-            old_cache.unlink()
+        def legacy_cleanup():
+            """Clean up old tokens."""
+            old_cache = Path(hass.config.path(".ring_cache.pickle"))
+            if old_cache.is_file():
+                old_cache.unlink()
 
-    await hass.async_add_executor_job(legacy_cleanup)
+        await hass.async_add_executor_job(legacy_cleanup)
```

Comment:
No, this actually introduces an error. Line 54 is *not* part of the conditional starting in line 46.